e-commerce-ui/src/app/login/login.component.ts

```diff
@@ -1,20 +1,40 @@
+import { HttpClient } from '@angular/common/http';
 import { Component } from '@angular/core';
-import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
-  styleUrls: ['./login.component.scss']
+  styleUrls: ['./login.component.scss'],
 })
 export class LoginComponent {
-
   emailForm: FormGroup;
   emailControl: FormControl;
+  errorMessage: string = '';
 
-  constructor() {
-    this.emailControl = new FormControl('example@example.com', [Validators.required, Validators.email]);
+  constructor(private http: HttpClient) {
+    this.emailControl = new FormControl('example@example.com', [
+      Validators.required,
+      Validators.email,
+    ]);
     this.emailForm = new FormGroup({
-      email: this.emailControl
+      email: this.emailControl,
     });
   }
-}
\ No newline at end of file
+
+  onLogin() {
+    this.http
+      .get(
+        `http://localhost:8080/api/v1/users/check-email?email=${this.emailForm.value.email}`
+      )
```

Comment:
Move the login components into a `auth` folder. Create an authentication service (ex,. `auth.service.ts`) and put all authentication-related HTTP requests in the service file. You can generate the service files using ng command `ng g s auth/auth`